images/router/haproxy/reload-haproxy

```diff
@@ -33,7 +33,7 @@ function haproxyHealthCheck() {
     while true; do
       local statusline=$(echo $'PROXY UNKNOWN\r\nGET / HTTP/1.1\r\n' | socat tcp-connect:localhost:${port} stdio | head -1)
 
-      if [[ "$statusline" == *" 503 "* ]]; then
+      if [ "$statusline" == "503" || "$statusline" == "404"]; then
```

Comment:
You're missing a space before the closing bracket.  Also, please use double-braces (Bash-style, not POSIX style).  
```suggestion
      if [[ "$statusline" = "503" || "$statusline" = "404" ]]; then
```

Reply:
Here, I think you need the `== *" 503 "*` format

Reply:
Ahh, this one does need globbing.  Somehow the `*` got transferred from here (where it is needed) to the check below (where it is not).  The `echo ... | socat ... | head -1` command will get something like "HTTP/1.0 404 Error", so we need to use globbing (`*" 404 "*`).

Reply:
Below, where `curl ... -w "%{http_code}"` is used, the result will be just the code: either "404" or "503", so a simple string equality check suffices there.